### ISSUE TITLE
wasm target has trim warnings

### ISSUE THREAD

**Issue by attilah** [corvus-dotnet/Corvus.JsonSchema#681]
I am using the library in a project that is targeting WASM with .NET 9.0/10.0 and since WASM target includes trimming I see 3 warnings about it:

```
ILLink : Trim analysis warning IL2026: Corvus.Json.JsonValueExtensions.Serialize<TValue>(TValue): Using member 'System.Text.Json.JsonSerializer.Serialize(!!0, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. JSON serialization and deserialization might require types that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.

ILLink : Trim analysis warning IL2026: Corvus.Json.NumericTypeExtensions.TryGetInt128WithFallbacks(JsonElement, Int128&): Using member 'System.Text.Json.JsonSerializer.Deserialize(JsonElement, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. JSON serialization and deserialization might require types that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.

ILLink : Trim analysis warning IL2026: Corvus.Json.NumericTypeExtensions.TryGetUInt128WithFallbacks(JsonElement, UInt128&): Using member 'System.Text.Json.JsonSerializer.Deserialize(JsonElement, JsonSerializerOptions)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. JSON serialization and deserialization might require types that cannot be statically analyzed. Use the overload that takes a JsonTypeInfo or JsonSerializerContext, or make sure all of the required types are preserved.
``` 

It would be nice it trimming would be fully supported so WASM targets would work without warnings.

**Comment by mwadams** [comment]
Agreed.

Irritatingly, this is purely because of serialisation. We will figure it out!

**Comment by mwadams** [comment]
## Implementation Plan (v2)

### Problem
3 trim warnings when targeting WASM with AOT/trimming — all from `System.Text.Json.JsonSerializer` APIs with `[RequiresUnreferencedCode]`.

### Fix 1: `JsonValueExtensions.Serialize<TValue>` (lines 101-123)

**Problem:** `JsonSerializer.Serialize(value)` and `JsonSerializer.Serialize(value, options)` require reflection.

**Fix:** V4 types implement `IJsonValue.WriteTo(Utf8JsonWriter)`. Use that directly:

```csharp
// Parameterless overload
if (value.HasJsonElementBacking)
    return value.AsJsonElement.GetRawText();

using var buffer = new ArrayBufferWriter<byte>();
using var writer = new Utf8JsonWriter(buffer);
value.WriteTo(writer);
writer.Flush();
return Encoding.UTF8.GetString(buffer.WrittenSpan);

// Options overload — map to JsonWriterOptions
using var buffer = new ArrayBufferWriter<byte>();
using var writer = new Utf8JsonWriter(buffer, new JsonWriterOptions
{
    Encoder = options.Encoder,
    Indented = options.WriteIndented,
});
value.WriteTo(writer);
writer.Flush();
return Encoding.UTF8.GetString(buffer.WrittenSpan);
```

### Fix 2 & 3: `TryGetInt128WithFallbacks` / `TryGetUInt128WithFallbacks` (lines 273-290, 448-465)

**Problem:** `value.Deserialize<Int128>()` / `value.Deserialize<UInt128>()` require reflection.

**Fix:** Zero-allocation path using `JsonMarshal.GetRawUtf8Value` + `Int128.TryParse(ReadOnlySpan<byte>)`:

```csharp
public static bool TryGetInt128WithFallbacks(this JsonElement value, [NotNullWhen(true)] out Int128 result)
{
    if (value.ValueKind == JsonValueKind.Number
        && Int128.TryParse(JsonMarshal.GetRawUtf8Value(value), CultureInfo.InvariantCulture, out result))
    {
        return true;
    }

    result = default;
    return false;
}
```

Same pattern for `UInt128`.

**No `#if` guard needed** — `JsonMarshal.GetRawUtf8Value` is available unconditionally on the `System.Text.Json` 10.0.7 package. The `BUILDING_SOURCE_GENERATOR` gate in `LowAllocJsonUtils.cs` is only for the source generator context (which uses Roslyn''s STJ, not the package). The Int128/UInt128 code is already under `#if NET8_0_OR_GREATER` where both `JsonMarshal` and `IUtf8SpanParsable<Int128>` are available.

### Files to Modify

1. **`src-v4/Corvus.Json.ExtendedTypes/Corvus.Json/JsonValueExtensions.cs`** (lines 101-123)
   - Add `using System.Buffers; using System.Text;`
   - Replace both `JsonSerializer.Serialize` calls with `WriteTo` + `Utf8JsonWriter`

2. **`src-v4/Corvus.Json.ExtendedTypes/Corvus.Json/NumericTypeExtensions.cs`** (lines 273-290, 448-465)
   - Add `using System.Runtime.InteropServices; using System.Globalization;`
   - Replace `value.Deserialize<Int128/UInt128>()` with `JsonMarshal.GetRawUtf8Value` + `TryParse`

### Allocation profile

| Call site | Before | After |
|---|---|---|
| `Serialize()` (element-backed) | `GetRawText()` → 1 string | Same (unchanged path) |
| `Serialize()` (dotnet-backed) | `JsonSerializer.Serialize` → string + reflection metadata | `WriteTo` + `ArrayBufferWriter` + `GetString` → string only |
| `TryGetInt128/UInt128` | `Deserialize<T>()` → string + reflection metadata | `GetRawUtf8Value` (zero-alloc span) + `TryParse` → **zero allocation** |

**Comment by mwadams** [comment]
Completed: V4 trimming warnings from reflection-based `System.Text.Json.JsonSerializer` calls were removed by using `IJsonValue.WriteTo(Utf8JsonWriter)` and UTF-8 raw numeric parsing for Int128/UInt128 fallbacks.

Validation: focused V4 `JsonValueExtensionsTests`/`NumericTypeExtensions` tests and full `dotnet build Corvus.Text.Json.slnx` passed.
